client.js

```diff
@@ -174,7 +174,12 @@ S101Socket.prototype.connect = function (timeout = 2) {
             winston.debug('socket connected');
 
             self.keepaliveIntervalTimer = setInterval(() => {
-                self.sendKeepaliveRequest();
+                try {
```

Comment:
I would suggest to stop the keepalive on disconnect by clearing the interval.  Could you add it as part of this pull request.